### ISSUE TITLE
MSBuild no longer allows setting default configuration

### ISSUE THREAD

**Issue by Ghostbird** [dotnet/msbuild#8259]
### Issue Description

I have several project files that used: `<Configuration Condition=" '$(Configuration)' == '' ">Release</Configuration>`.
This stopped working, probably when I updated to MSBuild version 17.4.0+18d5aef85 for .NET

### Steps to Reproduce

1. `mkdir temp && cd temp && dotnet new console && code temp.csproj`
2. Insert this into the `PropertyGroup`:

    ```xml
    <Configuration Condition=" '$(Configuration)' == '' ">Release</Configuration>
    ```
3. `msbuild temp.csproj`

### Expected Behavior

A release build is made.

### Actual Behavior

A debug build is made:

```bash
~/workspace/temp$ dotnet build temp.csproj 
MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  temp -> /home/gijsbert/workspace/temp/bin/Debug/net7.0/temp.dll
```

### Analysis

It seems that the empty string comparison for the value of `'$(Configuration)'` no longer works.

### Versions & Configurations

I've included both the output of `dotnet build --version` and `msbuild --version` as a sanity check to ensure that there's not some installation problem where an unexpected version is run.
```bash
dotnet --version
7.0.101
dotnet build --version
MSBuild version 17.4.0+18d5aef85 for .NET
msbuild --version
MSBuild version 17.4.0+18d5aef85 for .NET
```

**Comment by Ghostbird** [comment]
Ah, so it's been like that for a while. I guess the documentation has just not been updated in several places, such as [this one](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild?view=vs-2022#BKMK_Properties).

I encountered this issue as I was porting an app from .NET Standard 2.1 Xamarin Forms to .NET 7.0 MAUI. This was accomplished by creating a new MAUI app and porting the source files as appropriate, so that checks out.

Thank you for your answer! I don't mind the new default. I just got really confused because the docs I read said that it's not set by default, and I was wondering if I made an error in the environment set-up.
